src/wallet/wallet.cpp

```diff
@@ -1571,11 +1571,18 @@ isminetype CWallet::IsMine(const CTxDestination& dest) const
 isminetype CWallet::IsMine(const CScript& script) const
 {
     AssertLockHeld(cs_wallet);
-    isminetype result = ISMINE_NO;
-    for (const auto& spk_man_pair : m_spk_managers) {
-        result = std::max(result, spk_man_pair.second->IsMine(script));
+
+    // Search the cache
+    const auto& it = m_cached_spks.find(script);
+    if (it != m_cached_spks.end()) {
+        // The cache is only used for descriptor wallets. If a script is in the cache, then it is automatically ISMINE_SPENDABLE.
+        return ISMINE_SPENDABLE;
```

Comment:
in "wallet: Use scriptPubKey cache in IsMine" (https://github.com/bitcoin/bitcoin/pull/26008/commits/3279e0e8078b92bc590d1ab65ba35f8d197edc12):

While this assumption is true today (and probably always will be), it feels like we are having CWallet make a decision that should be left up to the SPKMan. What if we searched the cache the same way as before and returned the max result from the SPKMan(s)?

```suggestion
        isminetype result = ISMINE_NO;
        for (const auto& spk : it->second) {
            result = std::max(result, spk->IsMine(script));
        }
        return result;
    }
```

I didn't compile this and test the benchmark, but I doubt it hurts the performance gains much (if any).

Reply:
Done